### PR TITLE
add responsename

### PR DESCRIPTION
**Pull request by matthieugomez**
Add `responsename`. Will be used by packages that print results of `RegressionModel`, such as `RegressionTables`.

### CODE REVIEW

File: src/StatsBase.jl [hunk 1]
```diff
@@ -188,10 +188,11 @@ export
     islinear,
     leverage,
     loglikelihood,
-    meanresponse,
     modelmatrix,
     mss,
     response,
+    responsename,
+    meanresponse,
```
Comment (nalimilan) [inline src/StatsBase.jl]: AFAICT names in this list are sorted lexicographically, so better keep the current order.

File: src/statmodels.jl [hunk 2]
```diff
@@ -294,6 +294,12 @@ Return the model response (a.k.a. the dependent variable).
 """
 response(obj::RegressionModel) = error("response is not defined for $(typeof(obj)).")
 
+"""
+    responsename(obj::StatisticalModel)
+Return the name of the model response (a.k.a. the dependent variable).
```
Comment (nalimilan) [inline src/statmodels.jl]:
```suggestion

Return the name of the model response (a.k.a. the dependent variable).
```

**Comment by kleinschmidt** [comment]
I certainly would like to move towards requiring models to do their own "formula bookkeeping" and add getter/setter methods to the StatsBase/StatsModels API.  I don't think that's incompatible with this proposal here, since we could just add a fallback `responsename(mod) = responsename(formula(mod))` or whatever the getter function ends up being called.

**Comment by nalimilan** [comment]
Yeah, that would be OK.

**Comment by matthieugomez** [comment]
updated.

**Pull request merged**